src/test/scala/com/mongodb/spark/config/ReadConfigSpec.scala

```diff
@@ -16,47 +16,95 @@
 
     readConfig should equal(expectedReadConfig)
   }
 
   it should "be creatable from SparkConfig" in {
-    val expectedReadConfig = ReadConfig("db", "collection", "secondary", Some("local"), 150) // scalastyle:ignore
+    val expectedReadConfig = ReadConfig("db", "collection", 150, 32, "ID", shardedConnectDirectly = true, shardedConnectToMongos = false,
+      ReadPreferenceConfig(ReadPreference.secondary()), ReadConcernConfig(ReadConcern.LOCAL))
 
     ReadConfig(sparkConf) should equal(expectedReadConfig)
   }
 
+  it should "use the URI for default values" in {
+    val uri =
+      "mongodb://localhost/db.collection?readPreference=secondaryPreferred&readPreferenceTags=dc:east,use:production&readPreferenceTags=&readconcernlevel=local"
+    val readConfig = ReadConfig(Map("uri" -> uri))
+
+    val expectedReadConfig = ReadConfig("db", "collection", 1000, 64, "_id", shardedConnectDirectly = false, shardedConnectToMongos = true,
+      ReadPreferenceConfig(ReadPreference.secondaryPreferred(List(
+        new TagSet(List(new Tag("dc", "east"), new Tag("use", "production")).asJava),
+        new TagSet()
+      ).asJava)),
+      ReadConcernConfig(ReadConcern.LOCAL))
+
+    readConfig should equal(expectedReadConfig)
+  }
+
+  it should "override URI values with named values" in {
+    val uri =
+      "mongodb://localhost/db.collection?readPreference=secondaryPreferred&readconcernlevel=local"
+    val readConfig = ReadConfig(Map("uri" -> uri, "readPreference.name" -> "primaryPreferred", "readConcern.level" -> "majority"))
+
+    val expectedReadConfig = ReadConfig("db", "collection", 1000, 64, "_id", shardedConnectDirectly = false, shardedConnectToMongos = true,
+      ReadPreferenceConfig(ReadPreference.primaryPreferred()), ReadConcernConfig(ReadConcern.MAJORITY))
+
+    readConfig should equal(expectedReadConfig)
+  }
+
   it should "be able mixin user parameters" in {
-    val expectedReadConfig = ReadConfig("db", "collection", "secondaryPreferred", Some("majority"), 200) // scalastyle:ignore
+    val expectedReadConfig = ReadConfig("db", "collection", 200, 20, "foo", shardedConnectDirectly = true, shardedConnectToMongos = false,
+      ReadPreferenceConfig(ReadPreference.secondaryPreferred(new TagSet(List(new Tag("dc", "east"), new Tag("use", "production")).asJava))),
+      ReadConcernConfig(ReadConcern.MAJORITY))
 
     ReadConfig(sparkConf).withOptions(
       Map(
-        "readPreference" -> "secondaryPreferred",
-        "readConcernLevel" -> "majority",
+        "maxChunkSize" -> "20",
+        "splitKey" -> "foo",
+        "shardedConnectDirectly" -> "true",
+        "shardedConnectToMongos" -> "false",
+        "readPreference.name" -> "secondaryPreferred",
+        "readPreference.tagSets" -> """[{dc: "east", use: "production"}]""",
+        "readConcern.level" -> "majority",
         "sampleSize" -> "200"
       )
     ) should equal(expectedReadConfig)
   }
 
   it should "be able to create a map" in {
```

Comment:
Nit: I was looking for parallels between ReadConfigSpec and WriteConfigSpec, and looks like this method is testing the same thing as WriteConfigSpec."round trip options".  If I'm right, please rename one of the tests so that they're the same.